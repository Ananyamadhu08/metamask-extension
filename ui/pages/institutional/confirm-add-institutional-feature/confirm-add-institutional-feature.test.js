import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import mockState from '../../../../test/data/mock-state.json';
import ConfirmAddInstitutionalFeature from '.';

const mockRemoveConnectInstitutionalFeature = jest
  .fn()
  .mockReturnValue({ type: 'TYPE' });

let mockSetComplianceAuthData = jest.fn().mockReturnValue({ type: 'TYPE' });

jest.mock('../../../store/institutional/institution-background', () => ({
  mmiActionsFactory: () => ({
    setComplianceAuthData: mockSetComplianceAuthData,
    removeConnectInstitutionalFeature: mockRemoveConnectInstitutionalFeature,
  }),
}));

const connectRequests = [
  {
    labels: [
      {
        key: 'service',
        value: 'test',
      },
    ],
    origin: 'origin',
    token: {
      projectName: 'projectName',
      projectId: 'projectId',
      clientId: 'clientId',
    },
  },
];

const props = {
  history: {
    push: jest.fn(),
  },
};

const render = ({ newState } = {}) => {
  const state = {
    ...mockState,
    metamask: {
      provider: {
        type: 'test',
      },
      institutionalFeatures: {
        complianceProjectId: '',
        connectRequests,
      },
      preferences: {
        useNativeCurrencyAsPrimaryCurrency: true,
      },
      ...newState,
    },
  };
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(state);

  return renderWithProvider(
    <ConfirmAddInstitutionalFeature {...props} />,
    store,
  );
};

describe('Confirm Add Institutional Feature', function () {
  it('opens confirm institutional sucessfully', () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
    expect(
      screen.getByText(`Id: ${connectRequests[0].token.projectId}`),
    ).toBeInTheDocument();
  });

  it('runs removeConnectInstitutionalFeature on cancel click', () => {
    render();
    fireEvent.click(screen.queryByText('Cancel'));
    expect(mockRemoveConnectInstitutionalFeature).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledTimes(1);
  });

  it('runs setComplianceAuthData on confirm click', () => {
    render();
    fireEvent.click(screen.queryByText('Confirm'));
    expect(mockSetComplianceAuthData).toHaveBeenCalledTimes(1);
  });

  it('handles error', () => {
    mockSetComplianceAuthData = jest
      .fn()
      .mockReturnValue(new Error('Async error message'));
    const { container } = render();
    fireEvent.click(screen.queryByText('Confirm'));
    expect(container).toMatchSnapshot();
  });

  it('does not render without connectRequest', () => {
    const newState = {
      institutionalFeatures: {
        connectRequests: [],
      },
    };
    const { container } = render({ newState });
    expect(container).toMatchSnapshot();
  });
});
