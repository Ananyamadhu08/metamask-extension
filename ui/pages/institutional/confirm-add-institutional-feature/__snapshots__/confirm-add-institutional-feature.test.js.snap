// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Add Institutional Feature does not render without connectRequest 1`] = `<div />`;

exports[`Confirm Add Institutional Feature handles error 1`] = `
<div>
  <div
    class="box page-container box--flex-direction-row"
  >
    <div
      class="box page-container__header box--flex-direction-row"
    >
      <p
        class="box mm-text page-container__title mm-text--body-md mm-text--color-text-default box--flex-direction-row"
      >
        Institutional Features
      </p>
      <p
        class="box mm-text page-container__subtitle mm-text--body-md mm-text--color-text-default box--flex-direction-row"
      >
        The page at origin would like to authorise the following project’s compliance settings in MetaMask Institutional
      </p>
    </div>
    <div
      class="box page-container__content box--flex-direction-row"
    >
      <p
        class="box mm-text mm-text--body-sm mm-text--color-text-default box--margin-top-3 box--margin-right-8 box--margin-left-8 box--flex-direction-row"
      >
        Project Name
      </p>
      <p
        class="box mm-text mm-text--body-lg-medium mm-text--color-text-default mm-text--overflow-wrap-break-word box--margin-top-1 box--margin-right-8 box--margin-bottom-1 box--margin-left-8 box--flex-direction-row"
      >
        projectName
      </p>
      <p
        class="box mm-text mm-text--body-xs mm-text--color-text-muted mm-text--overflow-wrap-break-word box--margin-right-8 box--margin-left-8 box--flex-direction-row"
      >
        Id
        : 
        projectId
      </p>
    </div>
    <p
      class="box mm-text mm-text--body-md mm-text--text-align-center mm-text--color-text-default box--margin-top-4 box--flex-direction-row"
    >
      Actions must be plain objects. Use custom middleware for async actions.
    </p>
    <div
      class="box page-container__footer box--flex-direction-row"
    >
      <footer>
        <button
          class="button btn--rounded btn-default btn--large"
          role="button"
          tabindex="0"
        >
          Cancel
        </button>
        <button
          class="button btn--rounded btn-primary btn--large"
          role="button"
          tabindex="0"
        >
          Confirm
        </button>
      </footer>
    </div>
  </div>
</div>
`;

exports[`Confirm Add Institutional Feature opens confirm institutional sucessfully 1`] = `
<div>
  <div
    class="box page-container box--flex-direction-row"
  >
    <div
      class="box page-container__header box--flex-direction-row"
    >
      <p
        class="box mm-text page-container__title mm-text--body-md mm-text--color-text-default box--flex-direction-row"
      >
        Institutional Features
      </p>
      <p
        class="box mm-text page-container__subtitle mm-text--body-md mm-text--color-text-default box--flex-direction-row"
      >
        The page at origin would like to authorise the following project’s compliance settings in MetaMask Institutional
      </p>
    </div>
    <div
      class="box page-container__content box--flex-direction-row"
    >
      <p
        class="box mm-text mm-text--body-sm mm-text--color-text-default box--margin-top-3 box--margin-right-8 box--margin-left-8 box--flex-direction-row"
      >
        Project Name
      </p>
      <p
        class="box mm-text mm-text--body-lg-medium mm-text--color-text-default mm-text--overflow-wrap-break-word box--margin-top-1 box--margin-right-8 box--margin-bottom-1 box--margin-left-8 box--flex-direction-row"
      >
        projectName
      </p>
      <p
        class="box mm-text mm-text--body-xs mm-text--color-text-muted mm-text--overflow-wrap-break-word box--margin-right-8 box--margin-left-8 box--flex-direction-row"
      >
        Id
        : 
        projectId
      </p>
    </div>
    
    <div
      class="box page-container__footer box--flex-direction-row"
    >
      <footer>
        <button
          class="button btn--rounded btn-default btn--large"
          role="button"
          tabindex="0"
        >
          Cancel
        </button>
        <button
          class="button btn--rounded btn-primary btn--large"
          role="button"
          tabindex="0"
        >
          Confirm
        </button>
      </footer>
    </div>
  </div>
</div>
`;
